File: ZeeClothsApp/src/screens/HomeScreen.js
import React, { useEffect, useRef } from 'react';
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  Image,
  TouchableOpacity,
  Dimensions,
  Animated,
  StatusBar,
} from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { BlurView } from 'expo-blur';
import { useDispatch, useSelector } from 'react-redux';
import Icon from 'react-native-vector-icons/MaterialIcons';

import { fetchProducts, fetchCategories } from '../store/slices/productSlice';
import CustomNavbar from '../components/CustomNavbar';
import ProductCard from '../components/ProductCard';
import CategoryCard from '../components/CategoryCard';
import { colors, fonts, spacing } from '../styles/theme';

const { width, height } = Dimensions.get('window');

export default function HomeScreen({ navigation }) {
  const dispatch = useDispatch();
  const { products, categories, featuredProducts, isLoading } = useSelector(state => state.products);
  
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const slideAnim = useRef(new Animated.Value(50)).current;

  useEffect(() => {
    dispatch(fetchProducts());
    dispatch(fetchCategories());
    
    // Animate on mount
    Animated.parallel([
      Animated.timing(fadeAnim, {
        toValue: 1,
        duration: 1000,
        useNativeDriver: true,
      }),
      Animated.timing(slideAnim, {
        toValue: 0,
        duration: 1000,
        useNativeDriver: true,
      }),
    ]).start();
  }, []);

  const navigateToShop = () => {
    navigation.navigate('Shop');
  };

  const navigateToProduct = (productId) => {
    navigation.navigate('ProductDetail', { productId });
  };

  const navigateToCategory = (categoryId) => {
    navigation.navigate('Shop', { categoryId });
  };

  return (
    <View style={styles.container}>
      <StatusBar barStyle="light-content" backgroundColor="#000000" />
      <CustomNavbar navigation={navigation} />
      
      <ScrollView 
        style={styles.scrollView}
        showsVerticalScrollIndicator={false}
        bounces={false}
      >
        {/* Hero Section */}
        <View style={styles.heroSection}>
          <Image
            source={{ uri: 'https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg' }}
            style={styles.heroBackground}
            resizeMode="cover"
          />
          <LinearGradient
            colors={['rgba(0,0,0,0.3)', 'rgba(0,0,0,0.7)']}
            style={styles.heroOverlay}
          />
          
          <Animated.View 
            style={[
              styles.heroContent,
              {
                opacity: fadeAnim,
                transform: [{ translateY: slideAnim }],
              }
            ]}
          >
            <Text style={styles.heroText}>
              Experience premium menswear crafted with sustainable materials and minimalist design.
            </Text>
            <TouchableOpacity style={styles.heroButton} onPress={navigateToShop}>
              <Text style={styles.heroButtonText}>EXPLORE COLLECTION</Text>
            </TouchableOpacity>
          </Animated.View>
          
          <View style={styles.bottomWords}>
            <Text style={styles.bottomWord}>Precision.</Text>
            <Text style={styles.bottomWord}>Comfort.</Text>
            <Text style={[styles.bottomWord, styles.accentWord]}>Style.</Text>
          </View>
        </View>

        {/* Discover Section */}
        <View style={styles.discoverSection}>
          <Text style={styles.sectionTitle}>DISCOVER</Text>
          <View style={styles.discoverGrid}>
            {categories.slice(0, 4).map((category, index) => (
              <CategoryCard
                key={category.id}
                category={category}
                onPress={() => navigateToCategory(category.id)}
                style={styles.categoryCard}
              />
            ))}
          </View>
        </View>

        {/* Future Fashion Section */}
        <View style={styles.futureFashionSection}>
          <View style={styles.futureFashionContent}>
            <Image
              source={{ uri: 'https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg' }}
              style={styles.futureFashionImage}
              resizeMode="cover"
            />
            <View style={styles.futureFashionText}>
              <Text style={styles.futureFashionTitle}>FIND YOUR FUTURE FASHION</Text>
              <Text style={styles.futureFashionDescription}>
                Step into tomorrow's style with our cutting-edge collection. Where innovation meets elegance, 
                we craft garments that don't just follow trends—they define them.
              </Text>
              <TouchableOpacity style={styles.futureFashionButton} onPress={navigateToShop}>
                <Text style={styles.futureFashionButtonText}>SHOP NOW</Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>

        {/* Featured Section */}
        <View style={styles.featuredSection}>
          <View style={styles.featuredContent}>
            {/* Main Featured Card */}
            <TouchableOpacity 
              style={styles.featuredMainCard}
              onPress={() => navigateToProduct(featuredProducts[0]?.id)}
            >
              <Image
                source={{ uri: featuredProducts[0]?.image_url || 'https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg' }}
                style={styles.featuredMainImage}
                resizeMode="cover"
              />
              <LinearGradient
                colors={['rgba(0,0,0,0.4)', 'rgba(0,0,0,0.7)']}
                style={styles.featuredMainOverlay}
              />
              <View style={styles.featuredMainContent}>
                <Text style={styles.featuredMainTitle}>FIND YOUR STYLE</Text>
                <Text style={styles.featuredMainDescription}>
                  Discover your unique fashion identity with our curated collection
                </Text>
              </View>
            </TouchableOpacity>

            {/* Side Cards */}
            <View style={styles.featuredSideCards}>
              {featuredProducts.slice(1, 5).map((product, index) => (
                <TouchableOpacity
                  key={product.id}
                  style={styles.featuredSideCard}
                  onPress={() => navigateToProduct(product.id)}
                >
                  <Image
                    source={{ uri: product.image_url }}
                    style={styles.featuredSideImage}
                    resizeMode="cover"
                  />
                  <Text style={styles.featuredSideText}>{product.name}</Text>
                </TouchableOpacity>
              ))}
            </View>
          </View>
        </View>

        {/* Brand Text Section */}
        <View style={styles.brandTextSection}>
          <Text style={styles.brandText}>ZEECLOTHS</Text>
        </View>

        {/* Horizontal Cards Section */}
        <View style={styles.horizontalCardsSection}>
          <ScrollView 
            horizontal 
            showsHorizontalScrollIndicator={false}
            contentContainerStyle={styles.horizontalCardsContainer}
          >
            {featuredProducts.map((product, index) => (
              <TouchableOpacity
                key={`${product.id}-${index}`}
                style={styles.horizontalCard}
                onPress={() => navigateToProduct(product.id)}
              >
                <Image
                  source={{ uri: product.image_url }}
                  style={styles.horizontalCardImage}
                  resizeMode="cover"
                />
                <View style={styles.horizontalCardContent}>
                  <Text style={styles.horizontalCardTitle}>{product.name}</Text>
                  <Text style={styles.horizontalCardPrice}>₹{Math.round(product.price)}</Text>
                </View>
              </TouchableOpacity>
            ))}
          </ScrollView>
        </View>

        {/* Footer */}
        <View style={styles.footer}>
          <View style={styles.footerContent}>
            <View style={styles.footerTop}>
              <Image
                source={{ uri: 'https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg' }}
                style={styles.footerImage}
                resizeMode="cover"
              />
              <View style={styles.footerTextContent}>
                <View style={styles.footerBrand}>
                  <Text style={styles.footerTitle}>ZEECLOTHS</Text>
                  <Text style={styles.footerDescription}>
                    Discover your unique fashion identity with our curated collection of premium menswear. 
                    We blend sustainable materials with minimalist design to create timeless pieces.
                  </Text>
                </View>
                <View style={styles.footerLinks}>
                  <View style={styles.footerColumn}>
                    <Text style={styles.footerSubtitle}>QUICK LINKS</Text>
                    <TouchableOpacity onPress={() => navigation.navigate('Home')}>
                      <Text style={styles.footerLink}>Home</Text>
                    </TouchableOpacity>
                    <TouchableOpacity onPress={() => navigation.navigate('Shop')}>
                      <Text style={styles.footerLink}>Collection</Text>
                    </TouchableOpacity>
                    <TouchableOpacity onPress={() => navigation.navigate('About')}>
                      <Text style={styles.footerLink}>About</Text>
                    </TouchableOpacity>
                    <TouchableOpacity onPress={() => navigation.navigate('Contact')}>
                      <Text style={styles.footerLink}>Contact</Text>
                    </TouchableOpacity>
                  </View>
                  <View style={styles.footerColumn}>
                    <Text style={styles.footerSubtitle}>CONTACT INFO</Text>
                    <Text style={styles.footerContactText}>Email: info@zeecloths.com</Text>
                    <Text style={styles.footerContactText}>Phone: +1 (555) 123-4567</Text>
                    <Text style={styles.footerContactText}>Address: 123 Fashion St, Style City</Text>
                  </View>
                </View>
              </View>
            </View>
            <View style={styles.footerBottom}>
              <View style={styles.socialSection}>
                <Text style={styles.footerSubtitle}>FOLLOW US</Text>
                <View style={styles.socialLinks}>
                  <Text style={styles.socialLink}>Instagram</Text>
                  <Text style={styles.socialLink}>Facebook</Text>
                  <Text style={styles.socialLink}>Twitter</Text>
                </View>
              </View>
              <Text style={styles.copyright}>© 2024 ZEECLOTHS. All rights reserved.</Text>
            </View>
          </View>
        </View>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.black,
  },
  scrollView: {
    flex: 1,
  },
  heroSection: {
    height: height,
    position: 'relative',
  },
  heroBackground: {
    position: 'absolute',
    width: '100%',
    height: '100%',
  },
  heroOverlay: {
    position: 'absolute',
    width: '100%',
    height: '100%',
  },
  heroContent: {
    position: 'absolute',
    left: '4%',
    bottom: 140,
    maxWidth: 320,
  },
  heroText: {
    fontFamily: fonts.regular,
    fontSize: 16,
    lineHeight: 24,
    color: colors.white,
    marginBottom: 20,
  },
  heroButton: {
    backgroundColor: 'rgba(255, 255, 255, 0.1)',
    borderWidth: 1,
    borderColor: 'rgba(255, 255, 255, 0.3)',
    paddingVertical: 14,
    paddingHorizontal: 28,
    alignSelf: 'flex-start',
  },
  heroButtonText: {
    fontFamily: fonts.medium,
    fontSize: 14,
    color: colors.white,
    letterSpacing: 1,
  },
  bottomWords: {
    position: 'absolute',
    bottom: 16,
    left: 0,
    right: 0,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    gap: 12,
  },
  bottomWord: {
    fontFamily: fonts.regular,
    fontSize: 16,
    color: '#dcdcdc',
  },
  accentWord: {
    color: colors.cyan,
  },
  discoverSection: {
    backgroundColor: 'rgba(0, 0, 0, 0.8)',
    paddingVertical: 60,
    paddingHorizontal: spacing.large,
  },
  sectionTitle: {
    fontFamily: fonts.semiBold,
    fontSize: 36,
    color: colors.white,
    letterSpacing: 2,
    marginBottom: 40,
    textTransform: 'uppercase',
  },
  discoverGrid: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    gap: 20,
  },
  categoryCard: {
    width: (width - 80) / 2,
    height: 280,
    marginBottom: 20,
  },
  futureFashionSection: {
    backgroundColor: 'rgba(0, 0, 0, 0.8)',
    paddingVertical: 60,
    paddingHorizontal: spacing.large,
  },
  futureFashionContent: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 40,
  },
  futureFashionImage: {
    flex: 1,
    height: 400,
    borderRadius: 0,
  },
  futureFashionText: {
    flex: 1,
  },
  futureFashionTitle: {
    fontFamily: fonts.semiBold,
    fontSize: 28,
    color: colors.white,
    letterSpacing: 1,
    marginBottom: 20,
    textTransform: 'uppercase',
  },
  futureFashionDescription: {
    fontFamily: fonts.regular,
    fontSize: 14,
    lineHeight: 22,
    color: 'rgba(255, 255, 255, 0.8)',
    marginBottom: 30,
  },
  futureFashionButton: {
    backgroundColor: 'rgba(255, 255, 255, 0.1)',
    borderWidth: 1,
    borderColor: 'rgba(255, 255, 255, 0.3)',
    paddingVertical: 14,
    paddingHorizontal: 28,
    alignSelf: 'flex-start',
  },
  futureFashionButtonText: {
    fontFamily: fonts.medium,
    fontSize: 14,
    color: colors.white,
    letterSpacing: 1,
  },
  featuredSection: {
    backgroundColor: 'rgba(0, 0, 0, 0.8)',
    paddingVertical: 60,
    paddingHorizontal: spacing.large,
  },
  featuredContent: {
    flexDirection: 'row',
    gap: 20,
  },
  featuredMainCard: {
    flex: 1,
    height: 320,
    position: 'relative',
    backgroundColor: 'rgba(255, 255, 255, 0.1)',
  },
  featuredMainImage: {
    width: '100%',
    height: '100%',
  },
  featuredMainOverlay: {
    position: 'absolute',
    width: '100%',
    height: '100%',
  },
  featuredMainContent: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: [{ translateX: -100 }, { translateY: -50 }],
    alignItems: 'center',
    width: 200,
  },
  featuredMainTitle: {
    fontFamily: fonts.semiBold,
    fontSize: 18,
    color: colors.white,
    letterSpacing: 1,
    textAlign: 'center',
    marginBottom: 8,
  },
  featuredMainDescription: {
    fontFamily: fonts.regular,
    fontSize: 12,
    color: 'rgba(255, 255, 255, 0.8)',
    textAlign: 'center',
    lineHeight: 18,
  },
  featuredSideCards: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    gap: 10,
  },
  featuredSideCard: {
    width: (width * 0.5 - 60) / 2,
    height: 150,
    backgroundColor: 'rgba(255, 255, 255, 0.1)',
    position: 'relative',
  },
  featuredSideImage: {
    width: '100%',
    height: '100%',
  },
  featuredSideText: {
    position: 'absolute',
    bottom: 12,
    left: 12,
    fontFamily: fonts.medium,
    fontSize: 12,
    color: colors.white,
  },
  brandTextSection: {
    backgroundColor: 'rgba(0, 0, 0, 0.9)',
    paddingVertical: 80,
    alignItems: 'center',
    justifyContent: 'center',
  },
  brandText: {
    fontFamily: fonts.bold,
    fontSize: 48,
    color: colors.white,
    letterSpacing: 8,
    textAlign: 'center',
  },
  horizontalCardsSection: {
    backgroundColor: 'rgba(0, 0, 0, 0.8)',
    paddingVertical: 40,
  },
  horizontalCardsContainer: {
    paddingHorizontal: spacing.large,
    gap: 20,
  },
  horizontalCard: {
    width: 200,
    backgroundColor: 'rgba(255, 255, 255, 0.1)',
    borderWidth: 1,
    borderColor: 'rgba(255, 255, 255, 0.1)',
  },
  horizontalCardImage: {
    width: '100%',
    height: 250,
  },
  horizontalCardContent: {
    padding: 16,
  },
  horizontalCardTitle: {
    fontFamily: fonts.medium,
    fontSize: 14,
    color: colors.white,
    marginBottom: 4,
  },
  horizontalCardPrice: {
    fontFamily: fonts.semiBold,
    fontSize: 16,
    color: colors.cyan,
  },
  footer: {
    backgroundColor: colors.black,
    paddingVertical: 60,
    paddingHorizontal: spacing.large,
  },
  footerContent: {
    flex: 1,
  },
  footerTop: {
    flexDirection: 'row',
    gap: 40,
    marginBottom: 40,
  },
  footerImage: {
    flex: 1,
    height: 300,
    borderRadius: 8,
  },
  footerTextContent: {
    flex: 1,
    justifyContent: 'space-between',
  },
  footerBrand: {
    marginBottom: 30,
  },
  footerTitle: {
    fontFamily: fonts.semiBold,
    fontSize: 32,
    color: colors.white,
    letterSpacing: 1,
    marginBottom: 16,
  },
  footerDescription: {
    fontFamily: fonts.regular,
    fontSize: 14,
    lineHeight: 22,
    color: 'rgba(255, 255, 255, 0.8)',
  },
  footerLinks: {
    flexDirection: 'row',
    gap: 30,
  },
  footerColumn: {
    flex: 1,
  },
  footerSubtitle: {
    fontFamily: fonts.medium,
    fontSize: 12,
    color: colors.white,
    letterSpacing: 1,
    textTransform: 'uppercase',
    marginBottom: 16,
    borderBottomWidth: 1,
    borderBottomColor: colors.cyan,
    paddingBottom: 8,
    alignSelf: 'flex-start',
  },
  footerLink: {
    fontFamily: fonts.regular,
    fontSize: 14,
    color: 'rgba(255, 255, 255, 0.8)',
    marginBottom: 8,
    lineHeight: 20,
  },
  footerContactText: {
    fontFamily: fonts.regular,
    fontSize: 14,
    color: 'rgba(255, 255, 255, 0.8)',
    marginBottom: 6,
    lineHeight: 20,
  },
  footerBottom: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderTopWidth: 1,
    borderTopColor: 'rgba(255, 255, 255, 0.3)',
    paddingTop: 30,
  },
  socialSection: {
    alignItems: 'flex-start',
  },
  socialLinks: {
    flexDirection: 'row',
    gap: 20,
  },
  socialLink: {
    fontFamily: fonts.regular,
    fontSize: 12,
    color: colors.white,
  },
  copyright: {
    fontFamily: fonts.regular,
    fontSize: 12,
    color: colors.white,
    textAlign: 'right',
  },
});

File: app/src/main/java/com/zeecloths/app/presentation/screens/home/HomeScreen.kt
package com.zeecloths.app.presentation.screens.home

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.navigation.NavController
import coil.compose.AsyncImage
import com.zeecloths.app.R
import com.zeecloths.app.presentation.components.ZeeClothsTopBar
import com.zeecloths.app.presentation.navigation.Screen
import com.zeecloths.app.presentation.theme.*

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun HomeScreen(
    navController: NavController,
    viewModel: HomeViewModel = hiltViewModel()
) {
    val uiState by viewModel.uiState.collectAsState()
    
    Scaffold(
        topBar = {
            ZeeClothsTopBar(
                title = "ZEECLOTHS",
                navController = navController,
                showBackButton = false
            )
        }
    ) { paddingValues ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues),
            verticalArrangement = Arrangement.spacedBy(0.dp)
        ) {
            // Hero Section
            item {
                HeroSection(
                    onShopNowClick = {
                        navController.navigate(Screen.Shop.route)
                    }
                )
            }
            
            // Discover Section
            item {
                DiscoverSection(
                    categories = uiState.categories,
                    onCategoryClick = { categoryId ->
                        navController.navigate("${Screen.Shop.route}?category=$categoryId")
                    }
                )
            }
            
            // Future Fashion Section
            item {
                FutureFashionSection(
                    onShopNowClick = {
                        navController.navigate(Screen.Shop.route)
                    }
                )
            }
            
            // Featured Section
            item {
                FeaturedSection(
                    featuredProducts = uiState.featuredProducts,
                    onProductClick = { productId ->
                        navController.navigate(Screen.ProductDetail.createRoute(productId))
                    }
                )
            }
            
            // Brand Text Section
            item {
                BrandTextSection()
            }
            
            // Horizontal Cards Section
            item {
                HorizontalCardsSection(
                    products = uiState.featuredProducts,
                    onProductClick = { productId ->
                        navController.navigate(Screen.ProductDetail.createRoute(productId))
                    }
                )
            }
        }
    }
}

@Composable
private fun HeroSection(
    onShopNowClick: () -> Unit
) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(600.dp)
    ) {
        // Background Image
        AsyncImage(
            model = "https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg",
            contentDescription = "Hero Background",
            modifier = Modifier.fillMaxSize(),
            contentScale = ContentScale.Crop
        )
        
        // Dark Overlay
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(
                    Brush.radialGradient(
                        colors = listOf(
                            Color.Black.copy(alpha = 0.3f),
                            Color.Black.copy(alpha = 0.7f)
                        )
                    )
                )
        )
        
        // Content
        Column(
            modifier = Modifier
                .align(Alignment.BottomStart)
                .padding(32.dp)
                .padding(bottom = 80.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Text(
                text = "Experience premium menswear crafted with sustainable materials and minimalist design.",
                style = MaterialTheme.typography.bodyLarge.copy(
                    color = Color.White,
                    fontSize = 18.sp,
                    lineHeight = 28.sp
                ),
                modifier = Modifier.widthIn(max = 400.dp)
            )
            
            Button(
                onClick = onShopNowClick,
                colors = ButtonDefaults.buttonColors(
                    containerColor = Color.White.copy(alpha = 0.1f),
                    contentColor = Color.White
                ),
                modifier = Modifier.padding(top = 8.dp)
            ) {
                Text(
                    text = "EXPLORE COLLECTION",
                    style = MaterialTheme.typography.labelMedium.copy(
                        fontWeight = FontWeight.Medium,
                        letterSpacing = 1.sp
                    )
                )
            }
        }
        
        // Bottom Words
        Row(
            modifier = Modifier
                .align(Alignment.BottomCenter)
                .padding(16.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Text(
                text = "Precision.",
                style = MaterialTheme.typography.bodyLarge.copy(
                    color = Color.White.copy(alpha = 0.8f),
                    fontSize = 16.sp
                )
            )
            Text(
                text = "Comfort.",
                style = MaterialTheme.typography.bodyLarge.copy(
                    color = Color.White.copy(alpha = 0.8f),
                    fontSize = 16.sp
                )
            )
            Text(
                text = "Style.",
                style = MaterialTheme.typography.bodyLarge.copy(
                    color = ZeeCyan,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Medium
                )
            )
        }
    }
}

@Composable
private fun DiscoverSection(
    categories: List<com.zeecloths.app.domain.model.Category>,
    onCategoryClick: (Int) -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.Black.copy(alpha = 0.8f))
            .padding(32.dp)
    ) {
        Text(
            text = "DISCOVER",
            style = MaterialTheme.typography.headlineLarge.copy(
                color = Color.White,
                fontWeight = FontWeight.SemiBold,
                fontSize = 48.sp,
                letterSpacing = 2.sp
            ),
            modifier = Modifier.padding(bottom = 40.dp)
        )
        
        LazyVerticalGrid(
            columns = GridCells.Fixed(2),
            horizontalArrangement = Arrangement.spacedBy(20.dp),
            verticalArrangement = Arrangement.spacedBy(20.dp),
            modifier = Modifier.height(600.dp)
        ) {
            items(categories.take(4)) { category ->
                CategoryCard(
                    category = category,
                    onClick = { onCategoryClick(category.id) }
                )
            }
        }
    }
}

@Composable
private fun CategoryCard(
    category: com.zeecloths.app.domain.model.Category,
    onClick: () -> Unit
) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(280.dp)
            .clickable { onClick() }
    ) {
        AsyncImage(
            model = category.imageUrl ?: "https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg",
            contentDescription = category.name,
            modifier = Modifier.fillMaxSize(),
            contentScale = ContentScale.Crop
        )
        
        Text(
            text = category.name.uppercase(),
            style = MaterialTheme.typography.titleMedium.copy(
                color = Color.White,
                fontWeight = FontWeight.Medium,
                letterSpacing = 1.sp
            ),
            modifier = Modifier
                .align(Alignment.BottomStart)
                .padding(16.dp)
        )
    }
}

@Composable
private fun FutureFashionSection(
    onShopNowClick: () -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.Black.copy(alpha = 0.8f))
            .padding(32.dp),
        horizontalArrangement = Arrangement.spacedBy(60.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Image
        AsyncImage(
            model = "https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg",
            contentDescription = "Future Fashion",
            modifier = Modifier
                .weight(1f)
                .height(400.dp)
                .clip(RoundedCornerShape(0.dp)),
            contentScale = ContentScale.Crop
        )
        
        // Text Content
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(24.dp)
        ) {
            Text(
                text = "FIND YOUR FUTURE FASHION",
                style = MaterialTheme.typography.headlineLarge.copy(
                    color = Color.White,
                    fontWeight = FontWeight.SemiBold,
                    fontSize = 36.sp,
                    letterSpacing = 1.sp
                )
            )
            
            Text(
                text = "Step into tomorrow's style with our cutting-edge collection. Where innovation meets elegance, we craft garments that don't just follow trends—they define them.",
                style = MaterialTheme.typography.bodyLarge.copy(
                    color = Color.White.copy(alpha = 0.8f),
                    lineHeight = 28.sp
                )
            )
            
            Button(
                onClick = onShopNowClick,
                colors = ButtonDefaults.buttonColors(
                    containerColor = Color.White.copy(alpha = 0.1f),
                    contentColor = Color.White
                )
            ) {
                Text(
                    text = "SHOP NOW",
                    style = MaterialTheme.typography.labelMedium.copy(
                        fontWeight = FontWeight.Medium,
                        letterSpacing = 1.sp
                    )
                )
            }
        }
    }
}

@Composable
private fun FeaturedSection(
    featuredProducts: List<com.zeecloths.app.domain.model.Product>,
    onProductClick: (Int) -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.Black.copy(alpha = 0.8f))
            .padding(32.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(20.dp)
        ) {
            // Main Featured Card
            if (featuredProducts.isNotEmpty()) {
                FeaturedMainCard(
                    product = featuredProducts.first(),
                    onClick = { onProductClick(featuredProducts.first().id) },
                    modifier = Modifier.weight(1f)
                )
            }
            
            // Side Cards Grid
            LazyVerticalGrid(
                columns = GridCells.Fixed(2),
                horizontalArrangement = Arrangement.spacedBy(10.dp),
                verticalArrangement = Arrangement.spacedBy(10.dp),
                modifier = Modifier
                    .weight(1f)
                    .height(320.dp)
            ) {
                items(featuredProducts.take(4)) { product ->
                    FeaturedSideCard(
                        product = product,
                        onClick = { onProductClick(product.id) }
                    )
                }
            }
        }
    }
}

@Composable
private fun FeaturedMainCard(
    product: com.zeecloths.app.domain.model.Product,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier
            .fillMaxWidth()
            .height(320.dp)
            .clickable { onClick() },
        colors = CardDefaults.cardColors(
            containerColor = Color.White.copy(alpha = 0.1f)
        )
    ) {
        Box(modifier = Modifier.fillMaxSize()) {
            AsyncImage(
                model = product.imageUrl ?: "https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg",
                contentDescription = product.name,
                modifier = Modifier.fillMaxSize(),
                contentScale = ContentScale.Crop
            )
            
            // Overlay
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(Color.Black.copy(alpha = 0.4f))
            )
            
            Column(
                modifier = Modifier
                    .align(Alignment.Center)
                    .padding(24.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Text(
                    text = "FIND YOUR STYLE",
                    style = MaterialTheme.typography.headlineSmall.copy(
                        color = Color.White,
                        fontWeight = FontWeight.SemiBold,
                        letterSpacing = 1.sp
                    ),
                    textAlign = TextAlign.Center
                )
                
                Text(
                    text = "Discover your unique fashion identity with our curated collection",
                    style = MaterialTheme.typography.bodyMedium.copy(
                        color = Color.White.copy(alpha = 0.8f),
                        textAlign = TextAlign.Center
                    ),
                    modifier = Modifier.padding(top = 8.dp)
                )
            }
        }
    }
}

@Composable
private fun FeaturedSideCard(
    product: com.zeecloths.app.domain.model.Product,
    onClick: () -> Unit
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .height(150.dp)
            .clickable { onClick() },
        colors = CardDefaults.cardColors(
            containerColor = Color.White.copy(alpha = 0.1f)
        )
    ) {
        Box(modifier = Modifier.fillMaxSize()) {
            AsyncImage(
                model = product.imageUrl ?: "https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg",
                contentDescription = product.name,
                modifier = Modifier.fillMaxSize(),
                contentScale = ContentScale.Crop
            )
            
            Text(
                text = product.categoryName?.uppercase() ?: "FASHION",
                style = MaterialTheme.typography.labelMedium.copy(
                    color = Color.White,
                    fontWeight = FontWeight.Medium
                ),
                modifier = Modifier
                    .align(Alignment.BottomStart)
                    .padding(12.dp)
            )
        }
    }
}

@Composable
private fun BrandTextSection() {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.Black.copy(alpha = 0.9f))
            .padding(vertical = 80.dp),
        contentAlignment = Alignment.Center
    ) {
        Text(
            text = "ZEECLOTHS",
            style = MaterialTheme.typography.displayLarge.copy(
                color = Color.White,
                fontWeight = FontWeight.Bold,
                fontSize = 72.sp,
                letterSpacing = 8.sp
            ),
            textAlign = TextAlign.Center
        )
    }
}

@Composable
private fun HorizontalCardsSection(
    products: List<com.zeecloths.app.domain.model.Product>,
    onProductClick: (Int) -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.Black.copy(alpha = 0.8f))
            .padding(vertical = 40.dp)
    ) {
        LazyRow(
            horizontalArrangement = Arrangement.spacedBy(20.dp),
            contentPadding = PaddingValues(horizontal = 32.dp)
        ) {
            items(products) { product ->
                HorizontalProductCard(
                    product = product,
                    onClick = { onProductClick(product.id) }
                )
            }
        }
    }
}

@Composable
private fun HorizontalProductCard(
    product: com.zeecloths.app.domain.model.Product,
    onClick: () -> Unit
) {
    Card(
        modifier = Modifier
            .width(250.dp)
            .height(350.dp)
            .clickable { onClick() },
        colors = CardDefaults.cardColors(
            containerColor = Color.White.copy(alpha = 0.1f)
        )
    ) {
        Column {
            AsyncImage(
                model = product.imageUrl ?: "https://images.pexels.com/photos/1040945/pexels-photo-1040945.jpeg",
                contentDescription = product.name,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(280.dp),
                contentScale = ContentScale.Crop
            )
            
            Column(
                modifier = Modifier.padding(16.dp)
            ) {
                Text(
                    text = product.name,
                    style = MaterialTheme.typography.titleMedium.copy(
                        color = Color.White,
                        fontWeight = FontWeight.Medium
                    )
                )
                
                Text(
                    text = "₹${product.price.toInt()}",
                    style = MaterialTheme.typography.bodyMedium.copy(
                        color = ZeeCyan,
                        fontWeight = FontWeight.SemiBold
                    ),
                    modifier = Modifier.padding(top = 4.dp)
                )
            }
        }
    }
}